### PR TITLE
Fix for 772 and 773

### LINKED ISSUE #772: AppID Changed
I'm still getting a similar error:
```
==> /home/steam/valheim/logs/auto-update.out <==
[Valheim][steam]: ###########################################################################
[Valheim][steam]: Valheim Server - Sun Jan 14 06:00:01 MST 2024
[ODIN][INFO]  - Checking for updates
[ODIN][INFO]  - steamcmd found in path

==> /home/steam/valheim/logs/auto-backup.out <==
[Valheim][steam]: Backup process complete! Created 20240114-060001-backup.tar.gz

==> /home/steam/valheim/logs/auto-update.out <==
thread 'main' panicked at src/odin/server/update.rs:157:3:
Invalid app info
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[Valheim][steam]: No update available
[Valheim][steam]: ###########################################################################
```

_Originally posted by @mew1033 in https://github.com/mbround18/valheim-docker/discussions/744#discussioncomment-8125861_

### LINKED ISSUE #773: Execing into docker fails with odin commands
docker exec -it valheim_valheim_1 gosu steam odin stop/status/etc

the game runs under the steam user but exec goes in via the root user.

_Originally posted by @mbround18 in https://github.com/mbround18/valheim-docker/discussions/771#discussioncomment-8120594_

### PR DESCRIPTION
**Pull request by mbround18**
# Description

## Contributions

- Fixes #772
- Fixes #773
- QOL, uses steam user as base user in final container
- QOL, correctly sets the home directory and working directory of the containers
- QOL, uses common utils shell script for both script files. following DRY format.

## Checklist

- [x] I added one or multiple labels which best describes this PR.
- [x] I have tested the changes locally.
- [x] This PR has a reviewer on it.
- [x] I have validated my changes in a docker container and on Ubuntu. (Only needed for Odin or Docker Changes)

### CODE REVIEW

**Comment by mbround18** [comment]
> [](https://github.com/mbround18/valheim-docker/commit/d24326735e0458249ddf14ac210f79bf8e566cc8):

Those look like successful logs, have you tried running the game server?

**Comment by mbround18** [comment]
Found a new funny one, server only launches with old app id but old app id is dead. So i switch the app id right before launch. building canary now to test

**Comment by mbround18** [comment]
Can confirm my server shows up in server browser 
image

**Comment by FooDeas** [comment]
I'm trying again with cd15943.

**Comment by mbround18** [comment]
I see server in ingame browser too with status 
image

**Comment by FooDeas** [comment]
cd15943 starts now! But entering the console with gosu (and maybe the cronjob?) doesn't work:
```
steam@295714bb5ba6:~/valheim$ /usr/sbin/gosu steam /bin/bash /home/steam/scripts/auto_update.sh
error: failed switching to "steam": operation not permitted
```
There is another problem with the updater now:
```
steam@295714bb5ba6:~/valheim$ odin update
[ODIN][INFO]  - Checking for updates
thread 'main' panicked at src/odin/server/process.rs:38:10:
Failed to get process exe
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by mbround18** [comment]
> thread 'main' panicked at src/odin/server/process.rs:38:10:

You shouldnt need gosu anymore, just exec into the container and it takes you in as steam user

**Comment by mbround18** [comment]
Alright @FooDeas  it was some formatting issues with the manifests, i fixed the brittleness with some regex. Arguably just as brittle but its the only option for right now. Sadly steam doesnt output json. 

image

**Comment by FooDeas** [comment]
> You shouldnt need gosu anymore, just exec into the container and it takes you in as steam user

That's what I thought and did. But there is the cron setup in the entrypoint.sh that won't work anymore.

File: src/scripts/entrypoint.sh [hunk 1]
```diff
@@ -121,10 +97,10 @@ setup_cron() {
     "BASH_ENV=/env.sh" \
     "${SCRIPT_PATH}" \
     "${LOG_LOCATION}" \
-    > "/etc/cron.d/${CRON_NAME}"
-  echo "" >> "/etc/cron.d/${CRON_NAME}"
+    | sudo tee "/etc/cron.d/${CRON_NAME}"
+  echo "" | sudo tee -a "/etc/cron.d/${CRON_NAME}"
```
Comment (mbround18) [inline src/scripts/entrypoint.sh:101]: @FooDeas this line sets the cron script as root, the gosu should still work for the crons. ill setup some crons for tonight and let em run

Comment (FooDeas) [reply src/scripts/entrypoint.sh:101]: Right. I thought from the wrong user context.

Comment (mbround18) [reply src/scripts/entrypoint.sh:101]: You were right on the call out! im reworking the cron system rn

Comment (FooDeas) [reply src/scripts/entrypoint.sh:101]: Interesting. I was confused by the "**sudo** tee" statements writing the cron files. Based on this, I wrongly concluded that it is now in the Steam user context. After your hint, I also would have expected that there would be no problems. 😮

**Comment by mbround18** [comment]
Alrrirght @FooDeas  thiissssss should work :) all features tested locally

**Comment by FooDeas** [comment]
Looks good! I can test it and give you feedback tomorrow, if you want.

**Comment by FooDeas** [comment]
It now works without any problems!

**Pull request merged**